butterfly-utilities/src/test/java/com/paypal/butterfly/utilities/xml/XmlXPathRetrieveTest.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Test
+    public void notFoundTest() {
+        final String XPATH_EXPRESSION = "/test";
+        XmlXPathRetrieve xpathRetrieve = new XmlXPathRetrieve(XPATH_EXPRESSION, XPathConstants.NODESET).relative("foo1.xml");
+        assertEquals(xpathRetrieve.getXPathExpressionString(), XPATH_EXPRESSION);
+        TUExecutionResult executionResult = xpathRetrieve.execution(transformedAppFolder, transformationContext);
+        assertEquals(executionResult.getType(), TUExecutionResult.Type.VALUE);
+        Object result = executionResult.getValue();
+        NodeList nodes = (NodeList) result;
+        assertNotNull(executionResult.getValue());
+        assertEquals(nodes.getLength(), 0);
+        assertEquals(xpathRetrieve.getDescription(), "Retrieve the XML data based on the given XPath query " + XPATH_EXPRESSION +  " and XML file foo1.xml");
+    }
+
+    @Test(expectedExceptions = TransformationDefinitionException.class)
+    public void returnTypeNotSupportedTest() {
+        new XmlXPathRetrieve("/deliverable-settings", XPathConstants.BOOLEAN);
+    }
+
+    @Test(expectedExceptions = TransformationDefinitionException.class)
+    public void nullTypeNotSupportedTest() {
+        new XmlXPathRetrieve("/deliverable-settings", null);
+    }
+
+    @Test(expectedExceptions = TransformationDefinitionException.class)
+    public void emptyXPathExpressionTest() {
+        new XmlXPathRetrieve("`", XPathConstants.NODESET);
+    }
+
+    @Test(expectedExceptions = TransformationDefinitionException.class)
+    public void nullXPathExpressionTest() {
+        new XmlXPathRetrieve(null, XPathConstants.NODESET);
+    }
+
+    @Test
+    public void valueTest() {
+        final String XPATH_EXPRESSION = "//*[name()='jaxrs:server']/@address";
+        XmlXPathRetrieve xmlXPathElement = new XmlXPathRetrieve(XPATH_EXPRESSION).relative("foo.xml");
+        assertEquals(xmlXPathElement.getXPathExpressionString(), XPATH_EXPRESSION);
+        TUExecutionResult executionResult = xmlXPathElement.execution(transformedAppFolder, transformationContext);
+        assertEquals(executionResult.getType(), TUExecutionResult.Type.VALUE);
+        assertNotNull(executionResult.getValue());
+        assertEquals(executionResult.getValue(), "/blah");
+        assertEquals(xmlXPathElement.getDescription(), "Retrieve the XML data based on the given XPath query " + XPATH_EXPRESSION +  " and XML file foo.xml");
+    }
+
+    @Test
+    public void nullTest() {
+        final String XPATH_EXPRESSION = "/foo/bar";
+        XmlXPathRetrieve xmlXPathElement = new XmlXPathRetrieve(XPATH_EXPRESSION).relative("foo.xml");
+        assertEquals(xmlXPathElement.getXPathExpressionString(), XPATH_EXPRESSION);
+        TUExecutionResult executionResult = xmlXPathElement.execution(transformedAppFolder, transformationContext);
+        assertEquals(executionResult.getType(), TUExecutionResult.Type.VALUE);
+        assertNotNull(executionResult.getValue());
```

Comment:
What is the value supposed to be here, a blank String? If so, please assert that.